lib/model.js

```diff
@@ -18,9 +19,19 @@ class Model {
           `Warning: 'type' property is restricted in database schemas: ${name}`
         );
       }
+      const first = Object.keys(entity)[0];
+      if (first === 'Projection') {
+        last.add([name, entity]);
+        continue;
+      }
+      const schema = new Schema(name, entity, [this]);
+      this.entities.set(name, schema);
+    }
+    for (const [name, entity] of last) {
       const schema = new Schema(name, entity, [this]);
       this.entities.set(name, schema);
     }
+    last.clear();
```

Comment:
This introduces a Model loading race condition, why not put `last` inside of `Model` as a separate internal field?